site/en/docs/extensions/reference/storage/index.md

```diff
@@ -81,50 +82,115 @@ tubes can be filled, and if they are filled when you put your message in, it get
 going to be delayed by anyone that puts into that tube enormous amounts of material.
 
 For details on the current limits of the storage API, and what happens when those limits are
-exceeded, please see the quota information for [sync][5] and [local][6].
+exceeded, please see the quota information for [sync][prop-sync] and [local][prop-local].
 
 ## Examples
 
-The following example checks for CSS code saved by a user on a form, and if found, stores it.
+If you're interested in tracking changes made to a data object, you can add a listener to its
+`onChanged` event. Whenever anything changes in storage, that event fires. Here's sample code to
+listen for saved changes:
 
 ```js
-function saveChanges() {
-  // Get a value saved in a form.
-  var theValue = textarea.value;
-  // Check that there's some code there.
-  if (!theValue) {
-    message('Error: No value specified');
-    return;
+//// background.js ////
+
+chrome.storage.onChanged.addListener(function (changes, namespace) {
+  for (let [key, { oldValue, newValue }] of Object.entries(changes)) {
+    console.log(
+      `Storage key "${key}" in namespace "${namespace}" changed.`,
+      `Old value was "${oldValue}", new value is "${newValue}".`
+    );
   }
-  // Save it using the Chrome extension storage API.
-  chrome.storage.sync.set({'value': theValue}, function() {
-    // Notify that we saved.
-    message('Settings saved');
-  });
-}
+});
 ```
 
-If you're interested in tracking changes made to a data object, you can add a listener to its
-`onChanged` event. Whenever anything changes in storage, that event fires. Here's sample code to
-listen for saved changes:
+We can take this idea even further. In this example we have an [options page][options-page] that
+allows the user to toggle a "debug mode" in the extension (implementation not shown here). Changes
+to this setting are immediately saved to sync storage by the options page and the background script uses `storage.onChanged` to apply the setting as soon as possible.
 
 ```js
-chrome.storage.onChanged.addListener(function(changes, namespace) {
-  for (var key in changes) {
-    var storageChange = changes[key];
-    console.log('Storage key "%s" in namespace "%s" changed. ' +
-                'Old value was "%s", new value is "%s".',
-                key,
-                namespace,
-                storageChange.oldValue,
-                storageChange.newValue);
+//// options.html ////
+
+<script defer src="options.js"></script>
+<form id="optionsForm">
+  <label for="debug">
+    <input type="checkbox" name="debug" id="debug">
+    Enable debug mode
+  </label>
+</form>
+
+//// options.js ////
+
+// In-page cache of the user's options
+let options = {};
+
+// Initialize the form with the user's option settings
+chrome.storage.sync.get('options', (data) => {
+  Object.assign(options, data?.options);
+  optionsForm.debug.checked = data?.options?.debug;
```

Comment:
technically this would be `data.options?.debug`, _but_, since you've just assigned it to the local `options`: you can make this line:

```js
optionsForm.debug.checked = options.debug;

// or even better, in case it's undefined
optionsForm.debug.checked = Boolean(options.debug);
```

Reply:
I simplified another section to use the first pattern you suggested here. Also, while preparing the initial PR I tested to make sure `undefined` was handled the same as `false`. That said, I think your second suggestion is cleaner and clearer. I'll update both blocks to use this style.